src/presign/record.rs

```diff
@@ -132,4 +135,353 @@ impl PresignRecord {
         let signature_share = self.k * digest + x_projection * self.chi;
         Ok(SignatureShare::new(x_projection, signature_share))
     }
+
+    /// Convert private material into bytes.
+    ///
+    /// 🔒 This is intended for use by the calling application for secure
+    /// storage. The output of this function should be handled with care.
+    pub fn into_bytes(self) -> Vec<u8> {
+        // Format:
+        // RECORD TAG
+        // Curve point length in bytes (8 bytes)
+        // Curve point
+        // k randomness share length in bytes (8 bytes)
+        // k randomness share
+        // chi share length in bytes (8 bytes)
+        // chi share
+
+        let mut point = self.R.to_bytes();
+        let point_len = point.len().to_le_bytes();
+
+        let mut random_share = self.k.to_bytes();
+        let random_share_len = random_share.len().to_le_bytes();
+
+        let mut chi_share = self.chi.to_bytes();
+        let chi_share_len = chi_share.len().to_le_bytes();
+
+        let bytes = [
+            RECORD_TAG,
+            &point_len,
+            &point,
+            &random_share_len,
+            &random_share,
+            &chi_share_len,
+            &chi_share,
+        ]
+        .concat();
+
+        point.zeroize();
+        random_share.zeroize();
+        chi_share.zeroize();
+
+        bytes
+    }
+
+    /// Convert bytes into private material.
+    ///
+    /// 🔒 This is intended for use by the calling application for secure
+    /// storage. Do not use this method to create arbitrary instances of
+    /// [`PresignRecord`].
+    pub fn try_from_bytes(bytes: Vec<u8>) -> Result<Self> {
+        // Expected format:
+        // RECORD_TAG | key_len in bytes | key (big endian bytes)
+        //              | 8 bytes          | key_len bytes
+
+        let mut parser = ParseBytes::new(bytes);
+
+        // This little function ensures that
+        // 1. We can zeroize out the potentially-sensitive input bytes regardless of
+        //    whether parsing succeeded; and
+        // 2. We can log the error message once at the end, rather than duplicating it
+        //    across all the parsing code
+        let mut parse = || -> Result<PresignRecord> {
+            // Make sure the RECORD_TAG is correct.
+            let actual_tag = parser.take_bytes(RECORD_TAG.len())?;
+            if actual_tag != RECORD_TAG {
+                Err(CallerError::DeserializationFailed)?
+            }
+
+            // Parse the curve point
+            let point_len = parser.take_len()?;
+            let point_bytes = parser.take_bytes(point_len)?;
+            let point = CurvePoint::try_from_bytes(point_bytes)?;
+
+            // Parse the random share `k`
+            let random_share_len = parser.take_len()?;
+            let random_share_slice = parser.take_bytes(random_share_len)?;
+            let random_share_bytes: [u8; 32] = random_share_slice
```

Comment:
I see you are making a copy of the random share into `random_share_bytes`, should this variable be zeroized?

Reply:
Oh yeah, thanks. I had it loosely in my head that since we use `into` it would reuse the bytes to make the scalar, but it seems I can zeroize without any problem so I must not understand quite how that works.

Reply:
Right, if we are just dealing with a type that is just bytes, I assume Rust will just copy the bytes. So we end up having to zeroize.